### PR TITLE
[tqlotel] Avoid copying when filtering keys in KeepKeys.

### PR DESCRIPTION
**Pull request by kovrus**
**Description:** <Describe what has changed.>

Uses `AttributeMap#RemoveIf` introduced in [#4914](https://github.com/open-telemetry/opentelemetry-collector/pull/4914), also see [#4756](https://github.com/open-telemetry/opentelemetry-collector/issues/4756).

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector/issues/4756

**Testing:** <Describe what testing was performed and which tests were added.>
Existing unit tests.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: pkg/telemetryquerylanguage/functions/tqlotel/func_keep_keys.go [hunk 1]
```diff
@@ -33,16 +33,15 @@ func KeepKeys(target tql.GetSetter, keys []string) (tql.ExprFunc, error) {
...
 
 		if attrs, ok := val.(pcommon.Map); ok {
-			// TODO(anuraaga): Avoid copying when filtering keys https://github.com/open-telemetry/opentelemetry-collector/issues/4756
-			filtered := pcommon.NewMap()
-			filtered.EnsureCapacity(attrs.Len())
-			attrs.Range(func(key string, val pcommon.Value) bool {
+			attrs.RemoveIf(func(key string, value pcommon.Value) bool {
 				if _, ok := keySet[key]; ok {
-					filtered.Insert(key, val)
+					return false
```
Comment (dashpole) [inline pkg/telemetryquerylanguage/functions/tqlotel/func_keep_keys.go:38]: nit: you can just `return !ok` here

**Review by dashpole — APPROVED** [review]

**Comment by kovrus** [comment]
@dashpole should it be fine to skip the change log entry for this contribution? It is just a minor performance improvement.

**Comment by kovrus** [comment]
@TylerHelmuth can you please review the PR?

**Comment by kovrus** [comment]
Hi @dashpole can you take a look at it again and maybe merge it?

**Comment by dashpole** [comment]
We should make sure a codeowner takes a look

**Review by kentquirk — APPROVED** [review]
This looks good. Thank you!

**Comment by TylerHelmuth** [comment]
@kovrus I am on PTO but will review next week if still needed.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**